### PR TITLE
Feature/add devise

### PR DESCRIPTION
**Pull request by Alejuda**
Hello there! in this branch, we worked pair-to-pair with my coding partner @EmiLoKiJu and achieve the following:

- Remove current_user method in ApplicationController
- Install and set up devise
- User can:
  - register (that creates a new user)
  - Log in (If there is no user Logged in)
  - Log out (if there is a user Loged in)
  - Request a password change (since the mailer is not working yet, you can test it by clicking the link that you will see in the console when clicking Forgot password? link)
image
- Hashed password are stored in the database.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (laiifuu) [inline app/models/user.rb]: As mentioned in the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#:~:text=Ask%20for%20confirmation%20of%20email.), the application should ask for confirmation of email. You can achieve this by removing `:confirmable` from the commented out code, and adding it with the devise modules on line 4.

**Review by laiifuu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: 

## Hi team :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close


## Highlights
- [x] `Devise` set up correctly :+1: 
- [x] The registration and login views match the given wireframes :heavy_check_mark: 
- [x] A user can register and log in to the application :+1: 

## Required Changes ♻️
_Check the comments under the review :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and :computer: Happy coding :computer: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alejuda** [comment]
### fixed ♻️ 
Hi @laiifuu, Thank you for your review! It was really helpful 💯 
We already implemented the required email confirmation and can see in the console the email for confirmation as well:
----
image

**Review by laiifuu — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: :tada: 

## Hi @Alejuda

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :smiley: 🎉 

friends-joey

## Highlights
- [x] Necessary changes were made :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and :computer: Happy coding :computer: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**